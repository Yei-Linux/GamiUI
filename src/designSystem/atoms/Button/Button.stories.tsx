import { Meta } from "@storybook/react";
import {
  getListTemplate,
  stylesControl,
} from "../../../core/helpers/storybook";

import Button from ".";
import { options as optionsButton } from "./constants";
import { options as optionsStyles } from "../../../core/utils/constants";

const ListTemplate = getListTemplate(Button);

export default {
  title: "Atoms/Button",
  component: Button,
  argTypes: {
    children: { control: "text" },
    type: { control: { type: "select", options: optionsButton.type } },
    ...stylesControl,
  },
} as Meta;

export const TypesButton = ListTemplate.bind({});
TypesButton.args = {
  items: optionsButton.type.map((type) => ({ type })),
  field: 'type',
  children: "Text",
  width: "MEDIUM",
  heigth: "SMALL",
  shadow: "",
  border: "ROUNDED",
};

export const WidthButton = ListTemplate.bind({});
WidthButton.args = {
  items: optionsStyles.width.map((width) => ({ width })),
  field: 'width',
  children: "Text",
  heigth: "SMALL",
  shadow: "",
  border: "ROUNDED",
};
WidthButton.storyName = "Width Sizes";
