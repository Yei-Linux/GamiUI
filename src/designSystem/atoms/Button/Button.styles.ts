import { css } from "@emotion/react";
import styled from "@emotion/styled";
import { ButtonType } from "../../../core/domain/types";
import { colorLight } from "../../../styles/theme";
import { twinStyles } from "../../../styles/utilities/twinStyles";

const ButtonTypes = (typeStyle: ButtonType) => css`
  background: ${colorLight.button[typeStyle].bg};
  color: ${colorLight.button[typeStyle].color};
  border: ${colorLight.button[typeStyle].border};
`;

export const ButtonWrapper = styled.button<{
  typeStyle?: ButtonType;
  border?: string;
  shadow?: string;
  width?: string;
  heigth?: string;
  textAlign?: string;
  fontWeight?: string;
}>`
  border: none;
  outline: none;
  cursor: pointer;
  appearance: none;
  padding: 10px;

  ${(props: any) => twinStyles(props)};
  ${(props: any) => ButtonTypes(props.typeStyle)};
`;
