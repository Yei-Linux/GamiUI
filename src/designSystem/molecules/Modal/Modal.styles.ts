import styled from "@emotion/styled";
import { colorLight } from "../../../styles/theme";
import { twinStyles } from "../../../styles/utilities/twinStyles";

export const ModalWrapper = styled.div<{ visible: boolean }>`
  display: ${(props) => (props.visible ? "flex" : "none")};
`;

export const ModalMask = styled.div`
  background-color: rgba(0, 0, 0, 0.45);
  bottom: 0;
  left: 0;
  right: 0;
  top: 0;
  z-index: 1000;
  height: 100%;
  position: fixed;
`;

export const ModalDialog = styled.div`
  bottom: 0;
  left: 0;
  outline: 0;
  overflow: auto;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 1000;
  display: flex;
  justify-content: center;
  align-items: center;

  ${(props: any) => twinStyles(props)};
`;

export const ModalContainer = styled.div<{
  border?: string;
  shadow?: string;
}>`
  min-width: 500px;
  height: fit-content;
  background-color: ${colorLight.neutral.nine};
  ${(props: any) => twinStyles(props)};
`;

export const ModalHeader = styled.div`
  width: 100%;
  min-height: 50px;
  padding: 1rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-bottom: 1px solid ${colorLight.neutral.six};
`;

export const ModalFooter = styled.div`
  width: 100%;
  min-height: 50px;
  padding: 1rem;
  display: flex;
  align-items: center;
  border-top: 1px solid ${colorLight.neutral.six};
`;

export const ModalContent = styled.div`
  width: 100%;
  min-height: 50px;
  padding: 1rem;
`;
