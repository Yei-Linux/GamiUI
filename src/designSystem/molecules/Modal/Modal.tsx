import React, { useEffect, useState } from "react";
import { IGeneralProps } from "../../../core/domain/interfaces/IGeneralProps";
import { Left, Right } from "../../../styles/utilities/flex";
import { TitleContainer } from "../../../styles/utilities/text";
import Icon from "../../atoms/Icon";
import {
  ModalContainer,
  ModalContent,
  ModalDialog,
  ModalFooter,
  ModalHeader,
  ModalMask,
  ModalWrapper,
} from "./Modal.styles";

import { gsap } from "gsap";
import usePortal from "../../../hooks/usePortal";

interface IModal extends IGeneralProps {
  /**
   * Content Modal
   */
  children: React.ReactNode;
  /**
   * Header Modal
   */
  title: React.ReactNode;
  /**
   * Optional Footer
   */
  footer?: React.ReactNode;
  /**
   * Visible flag
   */
  visible: boolean;
  /**
   * Close Event
   */
  onClose: () => void;
}

const Header = ({
  title,
  handleClose,
}: {
  title: React.ReactNode;
  handleClose: () => any;
}) => (
  <ModalHeader>
    <Left>
      <TitleContainer>{title}</TitleContainer>
    </Left>
    <Right>
      <Icon fill="#7868e6" name="close" size="15px" onClick={handleClose} />
    </Right>
  </ModalHeader>
);

const Footer = ({ footer }: { footer: React.ReactNode }) => (
  <ModalFooter>{footer}</ModalFooter>
);

const Modal = ({
  children,
  title,
  footer,
  visible,
  onClose,
  ...args
}: IModal) => {
  let refDialog: any = null;
  let refContent: any = null;

  const {} = usePortal({ destinyElement: document.body });
  const [modalTween] = useState(gsap.timeline({ paused: true }));

  useEffect(() => {
    modalTween
      .to(refDialog, 0.35, { y: 0, autoAlpha: 1 })
      .from(
        refContent?.children,
        0.35,
        { y: 15, opacity: 0, stagger: 0.1 },
        "-=0.15"
      )
      .reverse();
  }, []);

  useEffect(() => {
    modalTween.reversed(!visible);
  }, [visible]);

  const handleCloseGsap = () => {
    modalTween.reverse();
    gsap.delayedCall(modalTween.duration(), onClose);
  };

  return (
    <ModalWrapper visible={visible}>
      <ModalMask />
      <ModalDialog ref={(e) => (refDialog = e)}>
        <ModalContainer {...args} ref={(e) => (refContent = e)}>
          <Header title={title} handleClose={handleCloseGsap} />
          <ModalContent>{children}</ModalContent>
          <Footer footer={footer} />
        </ModalContainer>
      </ModalDialog>
    </ModalWrapper>
  );
};

Modal.defaultProps = {
  shadow: "SMALL",
  border: "SMALL",
};

export default Modal;
