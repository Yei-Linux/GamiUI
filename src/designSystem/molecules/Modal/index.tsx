export { default } from './Modal'
export { options } from './constants'
