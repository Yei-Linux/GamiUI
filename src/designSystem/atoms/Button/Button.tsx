import React from "react";
import { IGeneralProps } from "../../../core/domain/interfaces/IGeneralProps";
import { ButtonType } from "../../../core/domain/types";
import { ButtonWrapper } from "./Button.styles";

export interface IButton extends IGeneralProps {
  /**
   * Content Button to show
   */
  children: React.ReactNode;
  /**
   * Button Type
   */
  type: ButtonType;
}

const Button = ({ children, type, ...args }: IButton) => {
  return (
    <ButtonWrapper {...args} typeStyle={type} type="submit">
      {children}
    </ButtonWrapper>
  );
};

Button.defaultProps = {
  type: "primary",
  border: "MEDIUM",
  heigth: "SMALL",
  width: "SMALL",
};

export default Button;
