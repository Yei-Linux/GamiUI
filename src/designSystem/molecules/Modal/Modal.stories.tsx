import React, { useState } from "react";
import { Meta } from "@storybook/react";
import { stylesControl } from "../../../core/helpers/storybook";

import Modal from ".";
import Button from "../../atoms/Button";

export default {
  title: "Molecules/Modal",
  component: Modal,
  args: { ...Modal.defaultProps },
  argTypes: {
    ...stylesControl,
  },
} as Meta;

export const Basic = () => {
  const [visible, setVisible] = useState(false);

  const onOpen = () => setVisible(true);

  const onClose = () => setVisible(false);

  return (
    <div>
      <Button
        onClick={onOpen}
        border="MEDIUM"
        heigth="SMALL"
        type="secondary"
        width="SMALL"
      >
        Show Modal
      </Button>
      <Modal visible={visible} onClose={onClose} title="Title">
        <p style={{ padding: "1rem" }}>Hola como estas</p>
      </Modal>
    </div>
  );
};
