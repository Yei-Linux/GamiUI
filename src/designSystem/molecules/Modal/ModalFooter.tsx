import React from 'react'
import * as S from './Modal.styles'

const Footer = ({ footer }: { footer: React.ReactNode }) => (
  <S.ModalFooter>{footer}</S.ModalFooter>
)

export default Footer
