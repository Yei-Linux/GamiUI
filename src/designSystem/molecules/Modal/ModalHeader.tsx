import Icon from 'designSystem/atoms/Icon'
import React from 'react'
import { Section, TitleContainer } from 'styles/utilities/commonComponent'
import * as S from './Modal.styles'

export interface IModalHeader {
  title: React.ReactNode
  handleClose: () => void
}

const ModalHeader = ({ title, handleClose }: IModalHeader) => (
  <S.ModalHeader>
    <Section>
      <TitleContainer>{title}</TitleContainer>
    </Section>
    <Section>
      <Icon fill="#7868e6" name="close" size="15px" onClick={handleClose} />
    </Section>
  </S.ModalHeader>
)

export default ModalHeader
