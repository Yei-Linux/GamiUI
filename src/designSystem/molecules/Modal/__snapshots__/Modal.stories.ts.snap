// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Molecules/Modal Basic 1`] = `
<div>
  <button
    className="css-1ysm1ze-ButtonWrapper ekfoxpm0"
    onClick={[Function]}
    type="submit"
    width="SMALL"
  >
    Show Modal
  </button>
  <div
    className="css-1zg4yv-ModalWrapper eje9c9k0"
  >
    <div
      className="css-oc6e3i-ModalMask eje9c9k1"
    />
    <div
      className="css-193xukr-ModalDialog eje9c9k2"
    >
      <div
        className="css-u01hii-ModalContainer eje9c9k3"
      >
        <div
          className="css-1ayc9ds-ModalHeader eje9c9k4"
        >
          <div
            className="css-a72bgn-Left e1skj02x0"
          >
            <div
              className="css-y29r7q-TitleContainer etn7vk90"
            >
              Title
            </div>
          </div>
          <div
            className="css-mgn85a-Right e1skj02x1"
          >
            <i
              className="css-6tw2qr-IconWrapper e1cr9qb30"
              onClick={[Function]}
            >
              <svg
                fill="#7868e6"
                height="15px"
                viewBox="0 0 329.26933 329"
                width="15px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m194.800781 164.769531 128.210938-128.214843c8.34375-8.339844 8.34375-21.824219 0-30.164063-8.339844-8.339844-21.824219-8.339844-30.164063 0l-128.214844 128.214844-128.210937-128.214844c-8.34375-8.339844-21.824219-8.339844-30.164063 0-8.34375 8.339844-8.34375 21.824219 0 30.164063l128.210938 128.214843-128.210938 128.214844c-8.34375 8.339844-8.34375 21.824219 0 30.164063 4.15625 4.160156 9.621094 6.25 15.082032 6.25 5.460937 0 10.921875-2.089844 15.082031-6.25l128.210937-128.214844 128.214844 128.214844c4.160156 4.160156 9.621094 6.25 15.082032 6.25 5.460937 0 10.921874-2.089844 15.082031-6.25 8.34375-8.339844 8.34375-21.824219 0-30.164063zm0 0"
                />
              </svg>
            </i>
          </div>
        </div>
        <div
          className="css-wu8ba1-ModalContent eje9c9k6"
        >
          <p
            style={
              Object {
                "padding": "1rem",
              }
            }
          >
            Hola como estas
          </p>
        </div>
        <div
          className="css-p70500-ModalFooter eje9c9k5"
        />
      </div>
    </div>
  </div>
</div>
`;
